File: 816-design-hashset/design-hashset.java
class MyHashSet {

    boolean[] dp=new boolean[1000001];
    public MyHashSet(){}
    
    public void add(int key) 
    {
        dp[key]=true;
    }
    
    public void remove(int key) 
    {
        dp[key]=false;
    }
    
    public boolean contains(int key) 
    {
        return dp[key]==true;
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.add(key);
 * obj.remove(key);
 * boolean param_3 = obj.contains(key);
 */

File: 816-design-hashset/design-hashset.cpp
class MyHashSet {
private:
    int CAP;
    vector<list<int>> buckets;
public:
    MyHashSet(){
        CAP = 10000;
        buckets = vector<list<int>>(CAP, list<int>{});
    }
    
    void add(int key){
        int index = key % CAP;
        auto& list = buckets[index];
        auto itr = find(list.begin(), list.end(), key);

        if(itr == list.end())
        list.push_back(key);
    }
    
    void remove(int key){
        int index = key % CAP;
        auto& list = buckets[index];
        auto itr = find(list.begin(), list.end(), key);

        if(itr != list.end())
        list.erase(itr);
    }
    
    bool contains(int key){
        int index = key % CAP;
        auto& list = buckets[index];
        auto itr = find(list.begin(), list.end(), key);

        return itr != list.end();
    }
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */